src/flint/fmpq.jl

```diff
@@ -792,7 +798,13 @@ end
 
 (a::FlintRationalField)() = fmpq(fmpz(0), fmpz(1))
 
-(a::FlintRationalField)(b::Rational) = fmpq(numerator(b), denominator(b))
+function (a::FlintRationalField)(b::Rational)
+   if denominator(b) < 0 # work around a Julia bug
```

Comment:
It would be useful to write here which bug (e.g. an example which fails, and maybe a corresponding test), so that eventually this work-around can be removed (when we will only support Julia versions where the bug is fixed).

Reply:
I can do that. Do you have a link to the PR or issue?

Reply:
RIght, referencing the julia issue is best: https://github.com/JuliaLang/julia/issues/32569. But I didn't check that this is always valid to just take the absolute value of the denominator. It looks like it though. Also, as this will now be an error in Julia, I'm not sure whether it's worth it to support a temporary set of possible invalid values, but there is no harm to do so either.

Reply:
I doubt they will backport this to all current version.

Reply:
I also doubt it, but eventually we won't support versions with the bug anymore (possibly in a very long time, but still).